src/validators/AppIdValidator.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+import {Ownable2Step} from "openzeppelin/contracts/access/Ownable2Step.sol";
+import {EIP712} from "openzeppelin/contracts/utils/cryptography/EIP712.sol";
+
+import {IMetadataValidator} from "../interfaces/IMetadataValidator.sol";
+import {IdRegistryLike} from "../interfaces/IdRegistryLike.sol";
+import {IdRegistry} from "../IdRegistry.sol";
+
+contract AppIdValidator is IMetadataValidator, Ownable2Step, EIP712 {
```

Comment:
What we're validating is that there is a signature from the entity that requested the signer, proving that they requested the signer. It may be an application, or it may be the user themselves. 

What this really is the the `RequestFidValidator` - the name is a little clunky, but directionally more accurate about what we are validating here.